pkg/operator/ceph/object/objectstore.go

```diff
@@ -56,129 +62,418 @@ type idType struct {
 	ID string `json:"id"`
 }
 
+type zoneGroupType struct {
+	MasterZoneID string     `json:"master_zone"`
+	IsMaster     string     `json:"is_master"`
+	Zones        []zoneType `json:"zones"`
+}
+
+type zoneType struct {
+	Name      string   `json:"name"`
+	Endpoints []string `json:"endpoints"`
+}
+
 type realmType struct {
 	Realms []string `json:"realms"`
 }
 
-func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
-	stores, err := getObjectStores(context)
+func deleteRealmAndPools(objContext *Context, spec cephv1.ObjectStoreSpec) error {
+	if spec.IsMultisite() {
+		// get list of endpoints not including the endpoint of the object-store for the zone
```

Comment:
nit: since these are such big code blocks, how about two sub-methods such as:
```
if spec.IsMultisite() {
   return deleteMultisiteRealmAndPools(objContext, spec)
}
return deleteSingleSiteRealmAndPools(objContext, spec)
}
```